src/System.CommandLine.Hosting/HostingExtensions.cs

```diff
@@ -43,16 +44,23 @@ public static class HostingExtensions
 
                 invocation.BindingContext.AddService(typeof(IHost), _ => host);
 
-                await host.StartAsync();
-
-                await next(invocation);
-
-                await host.StopAsync();
+                if (runAsDaemon)
+                {
+                    await next(invocation);
+                    await host.RunAsync();
```

Comment:
Maybe better using the cancellation token from the invocationContext so that we could graceful stop